### PR TITLE
171325 devops create cd pipeline

### PR DESCRIPTION
**Pull request by nerellavinod**
- Created CD pipeline for Dev, vNextIAT, vNextE2E environments.
- Azure resources created ( Webapp, Appinsight, KeyValut & EventHub ) using Terraform scripts and deployed on Dev environment.

### CODE REVIEW

**Review by amogh14893 — COMMENTED** [review]
Are these output variables being used for current scope of work. Re review the output variables across modules , retain those if getting used under current PBI scope.

File: Deployment/terraform_conditional_run.ps1
Comment (amogh14893) [inline Deployment/terraform_conditional_run.ps1:1]: Could you please rename 'mock' output variables with 'stub' . Additionally verify what all variables we need , apart from  api webapp , stub webapp and key vault URL there should not be any

File: azure-pipelines.yml
Comment (amogh14893) [inline azure-pipelines.yml:1]: Please review the output variables in  **DevDeployApp** Job.  if not required those can be removed and handled at stage level if possible.

**Review by amogh14893 — CHANGES_REQUESTED** [review]

File: Deployment/Modules/Webapp/main.tf
[the remark this answers is not in the document]
Comment (nerellavinod) [reply Deployment/Modules/Webapp/main.tf:1]: Resolved the comment

[the remark this answers is not in the document]
Comment (nerellavinod) [reply Deployment/Modules/Webapp/main.tf:1]: Resolved the comment.

File: Deployment/Modules/Webapp/outputs.tf
Comment (amogh14893) [inline Deployment/Modules/Webapp/outputs.tf:1]: Please correct output variables with prefix 'mock'. We are not using mock anywhere

File: Deployment/main.tf
Comment (amogh14893) [inline Deployment/main.tf:1]: Please correct webapp name variables for mock to stub.

**Review by jrippington — CHANGES_REQUESTED** [review]
Few minor corrections needed, thanks!

File: Deployment/Modules/Webapp/main.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  name                = var.mock_webapp_name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  service_plan_id     = azurerm_service_plan.app_service_plan.id
+  tags                = var.tags
+
+  site_config {
+    application_stack {    
+      current_stack  = "dotnet"
+      dotnet_version = "v6.0"
```
[the remark this answers is not in the document]
Comment (nerellavinod) [reply Deployment/Modules/Webapp/main.tf]: resolved now.

File: Deployment/Modules/Webapp/main.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  name                = var.name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  service_plan_id     = azurerm_service_plan.app_service_plan.id
+  tags                = var.tags
+
+  site_config {
+    application_stack {    
+      current_stack  = "dotnet"
+      dotnet_version = "v6.0"
```
[the remark this answers is not in the document]
Comment (nerellavinod) [reply Deployment/Modules/Webapp/main.tf]: resolved now.

**Review by jrippington — APPROVED** [review]
Looks good

**Pull request merged**